pkg/engine/virtualmachines.go

```diff
@@ -601,3 +605,33 @@ func addCustomTagsToVM(tags map[string]string, vm *compute.VirtualMachine) {
 		}
 	}
 }
+
+func associateAddonIdentitiesToVM(addonProfiles map[string]api.AddonProfile, virtualMachine *compute.VirtualMachine) {
```

Comment:
We need UT for this new func. Please add a test case for a nil `addonProfiles` and `virtualMachine`. Which means that we'll need to exit the func right away if a nil `virtualMachine` input is passed in.

Reply:
Thanks for reviewing! added unit test.